src/iqm/cirq_iqm/serialize.py

```diff
@@ -32,25 +160,36 @@ def serialize_circuit(circuit: Circuit) -> iqm_client.Circuit:
     Returns:
         data transfer object representing the circuit
     """
-    total_ops_list = [op for moment in circuit for op in moment]
+    total_ops_list = []
+    cc_prx_support = False
+    for moment in circuit:
+        for op in moment:
+            if isinstance(op, ClassicallyControlledOperation):
+                cc_prx_support = True
+            total_ops_list.append(op)
```

Comment:
Do you need to go over the circuit to check if it has cc_prx? Can't you just assume `cc_prx_support=True` and run the code below?

Reply:
You could just loop once over `circuit.all_operations()`, and convert the Cirq ops one by one to IQM ops, and then run the rest of the loop code below.

Reply:
Yeah, but if ccprx is found, you will loop over the circuit regardless. So I guess this is a speedup for circuits without ccprxs